### PR TITLE
Use rmw_allocate/rmw_free

### PR DESCRIPTION
**Pull request by allenh1**
Use rmw_allocate in place of malloc and rmw_free in place of free.

---

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3163/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/501/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/2520/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/3214/)

### CODE REVIEW

File: rmw_fastrtps_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -92,7 +93,7 @@ create_node(
     goto fail;
   }
 
-  node_handle = static_cast<rmw_node_t *>(malloc(sizeof(rmw_node_t)));
+  node_handle = static_cast<rmw_node_t *>(rmw_allocate(sizeof(rmw_node_t)));
```
Comment (mikaelarguedas) [inline rmw_fastrtps_cpp/src/rmw_node.cpp]: should this use `rmw_node_allocate` and `rmw_node_free` below ?

Comment (allenh1) [reply rmw_fastrtps_cpp/src/rmw_node.cpp]: Good catch

**Comment by allenh1** [comment]
@mikaelarguedas I changed the code as requested. Not sure why OS X was failing -- it looks unrelated (and I can't imagine this change should impact anything).


```
17:00:03 [ 13%] Built target geometry_msgs__py
17:00:03 make: *** [all] Error 2
17:00:03 make: INTERNAL: Exiting with 5 jobserver tokens available; should be 4!
17:00:03 
17:00:03 <== Command '. /Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs/cmake__build.sh && /usr/bin/make -j4 -l4' failed in '/Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs' with exit code '2'
17:00:03 SystemExit in geometry_msgs: <== Command '. /Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs/cmake__build.sh && /usr/bin/make -j4 -l4' failed in '/Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs' with exit code '2'
17:00:03 Traceback (most recent call last):
17:00:03   File "/Users/osrf/jenkins/workspace/ci_osx/ws/src/ament/ament_tools/ament_tools/verbs/build_pkg/cli.py", line 253, in run_command
17:00:03     subprocess.check_call(cmd, shell=True, cwd=cwd, env=build_action.env)
17:00:03   File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 291, in check_call
17:00:03     raise CalledProcessError(retcode, cmd)
17:00:03 subprocess.CalledProcessError: Command '. /Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs/cmake__build.sh && /usr/bin/make -j4 -l4' returned non-zero exit status 2.
17:00:03 
17:00:03 During handling of the above exception, another exception occurred:
17:00:03 
17:00:03 Traceback (most recent call last):
17:00:03   File "/Users/osrf/jenkins/workspace/ci_osx/ws/src/ament/ament_tools/ament_tools/verbs/build/cli.py", line 381, in process_in_parallel
17:00:03     result = done_future.result()
17:00:03   File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/concurrent/futures/_base.py", line 398, in result
17:00:03     return self.__get_result()
17:00:03   File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/concurrent/futures/_base.py", line 357, in __get_result
17:00:03     raise self._exception
17:00:03   File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/concurrent/futures/thread.py", line 55, in run
17:00:03     result = self.fn(*self.args, **self.kwargs)
17:00:03   File "/Users/osrf/jenkins/workspace/ci_osx/ws/src/ament/ament_tools/ament_tools/verbs/build_pkg/cli.py", line 280, in main
17:00:03     return run(opts, context)
17:00:03   File "/Users/osrf/jenkins/workspace/ci_osx/ws/src/ament/ament_tools/ament_tools/verbs/build_pkg/cli.py", line 331, in run
17:00:03     handle_build_action(on_build_ret, context)
17:00:03   File "/Users/osrf/jenkins/workspace/ci_osx/ws/src/ament/ament_tools/ament_tools/verbs/build_pkg/cli.py", line 270, in handle_build_action
17:00:03     run_command(build_action, context)
17:00:03   File "/Users/osrf/jenkins/workspace/ci_osx/ws/src/ament/ament_tools/ament_tools/verbs/build_pkg/cli.py", line 262, in run_command
17:00:03     sys.exit(msg)
17:00:03 SystemExit: <== Command '. /Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs/cmake__build.sh && /usr/bin/make -j4 -l4' failed in '/Users/osrf/jenkins/workspace/ci_osx/ws/build/geometry_msgs' with exit code '2'
```

The strangest thing is the line reading `17:00:03 make: INTERNAL: Exiting with 5 jobserver tokens available; should be 4!`. I'm going to go ahead and guess this is a very strange thing to happen, but I'll hold off on running the ci again until further review.

**Comment by mikaelarguedas** [comment]
OSX failure is not related to this change, see https://github.com/ros2/build_cop/issues/40

**Comment by allenh1** [comment]
@mikaelarguedas noted.

New CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3169/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/509/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/2526/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/3226/)

File: rmw_fastrtps_cpp/src/rmw_node.cpp [hunk 2]
```diff
@@ -270,11 +271,11 @@ rmw_destroy_node(rmw_node_t * node)
...
   delete impl->secondaryPubListener;
 
-  free(const_cast<char *>(node->name));
+  rmw_free(const_cast<char *>(node->name));
   node->name = nullptr;
-  free(const_cast<char *>(node->namespace_));
+  rmw_free(const_cast<char *>(node->namespace_));
   node->namespace_ = nullptr;
-  free(static_cast<void *>(node));
+  rmw_free(static_cast<void *>(node));
```
Comment (mikaelarguedas) [inline rmw_fastrtps_cpp/src/rmw_node.cpp]: rmw_node_free ?

Comment (allenh1) [reply rmw_fastrtps_cpp/src/rmw_node.cpp]: yep -- my bad.

**Pull request merged**